enzy_htp/_interface/pymol_interface.py

```diff
@@ -110,70 +227,266 @@ def config(self) -> PyMolConfig:
-    #     self.cmd.delete('all')
-
-    #     return sum(_eh_local['fc'])
-
-    # def get_sequence(self, fname: str, sele: str = '(all)') -> str:
-    #     """Method that gets the sequence for the specified sele in the
-    #     specified file. File must be a supported file type as listed in
-    #     PyMOLConfig. Checks if file exists and is supported type. If either
-    #     are not true then an error is logged and the program exits. DOES
-    #     NOT check if the sele is valid. Note that non-canonical residues will
-    #     be represented by a '?' in the sequence.
-    #     Args:
-    #         fname: The str() path to the file in question.
-    #         sele: The str() specifying the atom selection in PyMOL synatx. Default is '(all)'.
-    #     Returns:
-    #         A str() with the sequence of the sele as it would appear in a .fasta file..
-    #     """
-    #     if not Path(fname).exists():
-    #         _LOGGER.error(f"The supplied file '{fname}' does not exist. Exiting...")
-    #         exit(1)
-
-    #     if not self.supported_file_type(fname):
-    #         _LOGGER.error(
-    #             f"The supplied file '{fname}' has an unsupported extension. Exiting...")
-    #         exit(1)
-
-    #     self.cmd.delete('all')
-    #     self.cmd.load(fname)
-    #     lines: List[str] = self.cmd.get_fastastr(sele).splitlines()
-    #     self.cmd.delete('all')
-
-    #     lines = list(filter(lambda ll: ll[0] != '>', lines))
-    #     return ''.join(lines)
+
+    def de_protonate(self, molfile : str, new_file:str=None, session:pymol2.PyMOL=None) -> str:
+        """Method that deprotonates the structure supplied by molife, saving it to a new file. 
+        If no new_file is supplied, original file <name>.<ext> will be saved to <name>_deprotonated.<ext>
+
+        Args:
+            molfile: The name of the input file.
+            new_file: Where to save the deprotonated structure. Optional.
+            session : A pymol2.PyMOL() session to use. By default uses an internal session instance that resets each function call.
+    
+        Returns:
+            The name of the deprotonated structure.
+        """
+        self.check_pymol2_installed()
+
+        if not session:
+            self.session_.cmd.delete('all')
+            return self.de_protonate(
+                molfile,
+                new_file=new_file,
+                session=self.session_,
+            )
+
+
+        fs.check_file_exists( molfile )
+        
+        if not new_file:
+            fpath = Path(molfile )
+            new_file = fpath.parent / f"{fpath.stem}_deprotonated{fpath.suffix}"
+
+
+
+        session.cmd.delete( 'all' )
+        session.cmd.load( molfile )
+        session.cmd.remove( 'hydrogens' )
+        session.cmd.save( new_file )
+
+        session.cmd.delete('all')
+
+        return new_file
+
+    def rename_atoms(self, template: str, in_file: str) -> str:
```

Comment:
What is this? Could it be used to address [#117](https://github.com/ChemBioHTP/EnzyHTP/issues/117)

Reply:
This is specific to a problem I was having with the Rosetta paramterization script. I don't think it is applicable to the script you supplied. SDF's do not have atom names beyond element type and as a result there's a need to rename atoms from the BCL's conformer generation function.